### PR TITLE
127 chat functionality

### LINKED ISSUE #127: chat functionality
- Seen message
- Add user to chat
- Show chat members
- Typing...

### PR DESCRIPTION
**Pull request by maryanchuk04**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/users/maryanchuk04/projects/3)

## Code reviewers

- [ ] @Serhiy-dotcom 
- [ ] @Vova-Romaniuk 

### Second Level Review

- [ ] @Varvara-Maria 

## Summary of issue

Add new features to chat

## Summary of change

Create new files in server and client part
## Testing approach

ToDo

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: StudCity.Frontend/src/components/ChatModal/AddUserToChat.jsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const AddUserToChat = () => {
+	return (
+		<div>AddUserToChat</div>
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/components/ChatModal/AddUserToChat.jsx:5]: This is a future functional!

File: StudCity.Backend/StudCity.Core/Entities/Message.cs [hunk 2]
```diff
@@ -14,4 +14,8 @@ public class Message : BaseEntity
     public Room Room { get; set; }
 
     public Guid RoomId { get; set; }
+
```
Comment (maryanchuk04) [inline StudCity.Backend/StudCity.Core/Entities/Message.cs:17]: This is future functional!

File: StudCity.Frontend/src/app/features/chatsSlice.js [hunk 3]
```diff
@@ -58,12 +58,28 @@ const chatsSlice = createSlice({
...
+		},
 		loading: false,
 		hubConnection: null
 	},
 	reducers: {
 		addMessageAction: (state, action) => {
 			state.chat.messages = [...state.chat.messages, action.payload];
+		},
+		changeLastMessage: (state, action) => {
+			const index = state.userChats.findIndex((chat => chat.id === action.payload.roomId));
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/app/features/chatsSlice.js:73]: probably better to use find method instead of findIndex

**Review by Serhiy-dotcom — APPROVED** [review]
Well done!

**Review by Vova-Romaniuk — APPROVED** [review]

**Pull request merged**